### PR TITLE
Documentation for Kedro Viz Lite

### LINKED ISSUE #2059: Kedro Viz Lite Documentation and Telemetry
## Description

Documenting the `--lite` option in Kedro-Viz docs and adding Telemetry to track the usage

## Context

We introduced an option `--lite` on `kedro viz run` [here](https://github.com/kedro-org/kedro-viz/pull/1966), which will mock missing dependencies. This is a documentation/telemetry ticket for the same

## Possible Implementation

- Update README.md
- Update the doc [here](https://github.com/kedro-org/kedro-viz/blob/main/docs/source/kedro-viz_visualisation.md) with details on `kedro viz --lite`, limitations, user warnings etc
- Add any required filter in tracking the command run (this should already be tracked, but open for discussion during backlog grooming). 

## Possible Alternatives

Any other doc updates needed (discussion during backlog grooming)

## Checklist

- [x] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Resolves #2059 

## Development notes

* Updated README.md
* Updated the doc [here](https://github.com/kedro-org/kedro-viz/blob/main/docs/source/kedro-viz_visualisation.md) with details on kedro viz --lite, limitations, user warnings etc
* Updated [cli-docs](https://github.com/kedro-org/kedro-viz/blob/main/docs/source/cli-docs.md)

## QA notes

* All tests should pass

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: package/README.md [hunk 1]
```diff
@@ -157,6 +157,9 @@ Options:
                             To pass a nested dictionary as parameter, separate
                             keys by '.', example: param_group.param1:value1.
 
+  --lite                    An experimental flag to open Kedro-Viz with missing Kedro
+                            project dependencies
```
Comment (astrojuanlu) [inline package/README.md:161]:
```suggestion
  --lite                    An experimental flag to open Kedro-Viz with missing Kedro
                            project dependencies.
```

**Review by astrojuanlu — COMMENTED** [review]
Thanks @ravi-kumar-pilla , this looks good. Could you update `cli-docs.md` as well?

**Review by astrojuanlu — APPROVED** [review]

File: RELEASE.md [hunk 2]
```diff
@@ -10,11 +10,13 @@ Please follow the established format:
 ## Major features and improvements
 
 - Introduce `onActionCallback` prop in Kedro-Viz react component. (#2022)
+- Lightweight Kedro Viz Experimentation using AST. (#1966)
 
 ## Bug fixes and other changes
 
 - Fixes design issues in metadata panel. (#2009)
 - Fix missing run command in metadata panel for task nodes. (#2055)
+- Add docs for Kedro Viz Lite. (#2077)
```
Comment (rashidakanchwala) [inline RELEASE.md:19]: I think we don't need to add this

File: RELEASE.md [hunk 3]
```diff
@@ -10,11 +10,13 @@ Please follow the established format:
 ## Major features and improvements
 
 - Introduce `onActionCallback` prop in Kedro-Viz react component. (#2022)
+- Lightweight Kedro Viz Experimentation using AST. (#1966)
```
Comment (rashidakanchwala) [inline RELEASE.md:13]:
```suggestion
- Lightweight Kedro Viz Experimentation using AST. (#1966, #2077)
```

File: docs/source/cli-docs.md [hunk 4]
```diff
@@ -64,6 +64,9 @@ kedro viz run [OPTIONS]
 - `--params <params>`
   - Specify extra parameters for the Kedro Viz run. This option supports the same format as the `params` option in the Kedro CLI.
 
+- `--lite`                    
+  - An experimental flag to open Kedro-Viz with missing Kedro project dependencies.
```
Comment (rashidakanchwala) [inline docs/source/cli-docs.md:68]:
```suggestion
  - Run kedro-viz in lightweight mode without requiring all project dependencies.
```

File: docs/source/kedro-viz_visualisation.md [hunk 5]
```diff
@@ -63,6 +63,22 @@ If a visualisation panel opens up and a pipeline is not visible, refresh the vie
 
 To exit the visualisation, close the browser tab. To regain control of the terminal, enter `^+c` on Mac or `Ctrl+c` on Windows or Linux machines.
 
+## Visualise a Kedro project without installing project dependencies
+
+You can use the experimental `--lite` flag to visualise your pipelines without installing Kedro project dependencies. Add the flag to the command you use to start Kedro-Viz:
```
Comment (rashidakanchwala) [inline docs/source/kedro-viz_visualisation.md:68]:
```suggestion
You can use the --lite flag to run kedro-viz in lightweight mode, allowing you to visualize your pipelines without installing the full Kedro project dependencies. Add the flag to the command you use to start Kedro-Viz:
```

File: package/README.md [hunk 6]
```diff
@@ -157,6 +157,9 @@ Options:
                             To pass a nested dictionary as parameter, separate
                             keys by '.', example: param_group.param1:value1.
 
+  --lite                    An experimental flag to open Kedro-Viz with missing Kedro
```
Comment (rashidakanchwala) [inline package/README.md:160]:
```suggestion
  --lite                    Run `kedro-viz` in lightweight mode without requiring all Kedro project dependencies.
```

File: package/kedro_viz/launchers/cli/run.py [hunk 7]
```diff
@@ -81,7 +81,7 @@
 @click.option(
     "--lite",
     is_flag=True,
-    help="A flag to load an experimental light-weight Kedro Viz",
+    help="An experimental flag to open Kedro-Viz with missing Kedro project dependencies",
```
Comment (rashidakanchwala) [inline package/kedro_viz/launchers/cli/run.py:84]:
```suggestion
    help="Run `kedro-viz` in lightweight mode without requiring all project dependencies.",
```

Comment (astrojuanlu) [reply package/kedro_viz/launchers/cli/run.py:84]: So it's no longer "experimental"? 😄 

(Not that I'm pressing for it to be included again - we never defined what "experimental" means anyway...)

Comment (rashidakanchwala) [reply package/kedro_viz/launchers/cli/run.py:84]: I was thinking from the sense of -- does it help the user?

Comment (ravi-kumar-pilla) [reply package/kedro_viz/launchers/cli/run.py:84]: I agree we do not define what experimental means. I thought this would be something the user will proceed with caution considering the limitations if we tag it as experimental. Happy to discard as well. Also the lightweight mode is too wordy to understand for the user. Can we have - 
```
help="A flag to run Kedro-Viz without Kedro project dependencies"
```

Comment (rashidakanchwala) [reply package/kedro_viz/launchers/cli/run.py:84]: help="An experimental flag to run Kedro-Viz without Kedro project dependencies"

I think you both are right, we treat `kedro-viz --lite` as experimental. Once we see the feature's success, we can integrate it into our normal workflow and remove the `--lite`. Since this feature may not be permanent, adding it as an experimental option seems like a good idea.

**Review by rashidakanchwala — APPROVED** [review]
